src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java

```diff
@@ -672,6 +672,12 @@ protected final Object deserializeWithView(JsonParser p, DeserializationContext
                 SettableBeanProperty prop = _beanProperties.find(propName);
                 if (prop != null) {
                     if (!prop.visibleInView(activeView)) {
+                        // [databind#4108]: fields in other views to be considered as unknown properties
+                        if (ctxt.isEnabled(DeserializationFeature.FAIL_ON_UNEXPECTED_VIEW_PROPERTIES)){
+                            ctxt.reportInputMismatch(handledType(),
+                                String.format("Input mismatch while deserializing %s. Property '%s' is not part of current active view [%s]",
```

Comment:
Just realized one idea: should this suggest disabling of feature to allow?
(not 100% sure but might be helpful, esp. when 3.0 changes default to enabled)

Reply:
Right, not only we state the problem, but also inform a possible solution. Makes sense, will try to propose a improved version! 👍🏼